### DISCUSSION TITLE
Why are yup errors not being reported by `trigger` on an array field.

### DISCUSSION THREAD

**Discussion by chandu** [react-hook-form/react-hook-form#2438] [General] [answered]
The `trigger` method is not reporting that form is invalid when we are binding a model with an array property and if the property is not populated.
e.g: 
```
interface Person {
  name: string;
}

interface Persons {
  persons: Person[];
}

const personsValidator = yup.object<Persons>({
  persons: yup
    .array<Person>()
    .required()
    .min(1)
});

```
When I am binding the model mentioned above to RHF and am using `useFieldArray` hook to render the `persons` property.
When I call `trigger` without adding any person to the fields, RHF is validating the model successfully.
But the yup validator I have defined above marks persons field as required.

When I validate the model calling the yup validator directly, it is performing the validation correctly.

1. Click on `Validate Directly` button, `Direct Validation Message: ValidationError: persons is a required field` is shown
2. Click on 'Validate via RHF', `RHF Trigger Message: Is Valid: true` is shown. But I expect RHF's trigger to actually return an error saying `ValidationError: persons is a required field`
Why is RHF not repoting the error?

I expect the validation result either via yup or the rhf trigger (which is using the yupResolver) to be the same but here they are incosistent.

**Codesandbox**
https://codesandbox.io/s/dynamic-fields-validation-iiiso

**Browser ; Chrome
 - Version: 6.1.2/6.2.0

**Comment by chandu** [comment, marked as answer]
I guess the issue stems from this line:
https://github.com/react-hook-form/react-hook-form/blob/master/src/useForm.ts#L351 here we are only validating the fields that are in fieldRef.

**Reply by bluebill1049** [reply]
hey @Chandu this is expected, we only validate fields which are registered which are hosted in the fieldRef.

**Reply by chandu** [reply]
@bluebill1049 Thank you. How could one get to report all the validation errors returned by yup/resolver?

**Reply by bluebill1049** [reply]
you will probably need resolver to return your own errors

```ts
resolver: () => {
  const errors = yup.validate()...

  const customiseError = () => ...

  return {
    errors: customiseError(errors) // you can customise the error here
  }
}
```

**Reply by chandu** [reply]
@bluebill1049 Thanks.
But my use case is to ensure there is a specific entry in the array.
For e.g: 
Let's say I have the following result from yup (I am just printing the path of the error and the error message for simplicity)
```
{
	'persons[0].name': 'Cannot be empty'
	'persons[1].name': 'Cannot contains numbers'
	'persons': 'Atleast one entry by name BATMAN should be present'
}
```
How would I handle this scenario to display the individual person field level errors and also the errors at the persons array level?

**Reply by chandu** [reply]
I just noticed that my question is related to https://github.com/react-hook-form/react-hook-form/issues/2381 and https://github.com/react-hook-form/react-hook-form/issues/2379 with an added use case of being able to return errors at the array level (not just individual field of the array)

**Reply by bluebill1049** [reply]
I think you will probably have to set a separate error because persons are already used for array field.

`eg: setError('personsLength', 'a length error')`

**Reply by chandu** [reply]
Thanks @bluebill1049

**Reply by chandu** [reply]
@bluebill1049 Sorry for reviving an old discussion, to use your suggestion `setError('personsLength', 'a length error')` one would still have to use `yup.validate` explicitly and forego the `errors` object provided by `useForm`, is there a way to may be still pass the errors that are provided by yup.validate but are not included in `errors` object?

**Reply by bluebill1049** [reply]
hey @Chandu no worries, what do you mean by not include errors object?

**Reply by chandu** [reply]
In the example at: https://codesandbox.io/s/dynamic-fields-validation-iiiso I don't get the error: 'Atleast one entry by name BATMAN should be present' in the errors object.

How can I get the errors object populate this message and also avoid calling yup.validate explicitly?

**Reply by bluebill1049** [reply]
any chance you can add a rule for empty object check at yup? otherwise, you will have to use `shouldUnregister: false`

<img width="1262" alt="Screen Shot 2020-12-19 at 5 00 27 pm" src="https://user-images.githubusercontent.com/10513364/102682183-b3060b00-421b-11eb-9ed1-431ccd9339e4.png">

**Reply by chandu** [reply]
@bluebill1049 Thanks. I applied the changes you have suggested and can see that the persons object is now populated with the error message. However `trigger` method is still passing the test. Anyway to make it report the validation failed?
CSB: https://codesandbox.io/s/dynamic-fields-validation-forked-usx8b

**Reply by bluebill1049** [reply]
can you inspect why there are no errors? you can console log at the resolver against the schema. `trigger` is basically run through the `resolver`
